File: app/src/main/java/com/softaai/designpatterns/java/structural/bridge/Shape.java
package com.softaai.designpatterns.java.structural.bridge;

/**
 * Created by Amol Pawar on 03-04-2019.
 * softAai Apps
 */
public abstract class Shape {
   protected DrawAPI drawAPI;
   protected Shape(DrawAPI drawAPI){
       this.drawAPI = drawAPI;
   }

   public abstract void draw();
}

File: app/src/main/java/com/softaai/designpatterns/kotlin/structural/bridge/Shape.kt
package com.softaai.designpatterns.kotlin.structural.bridge


/**
 * Created by Amol Pawar on 03-04-2019.
 * softAai Apps
 */
abstract class Shape(drawAPI : DrawAPI) {
    protected lateinit var drawAPI:DrawAPI
    init{
        this.drawAPI = drawAPI
    }
    abstract fun draw()
}